libraries/AC_AutoTune/AC_AutoTune.h

```diff
@@ -90,21 +132,28 @@ class AC_AutoTune {
     void twitching_abort_rate(float angle, float rate, float angle_max, float meas_rate_min);
     void twitching_test_angle(float angle, float rate, float angle_target, float &meas_angle_min, float &meas_angle_max, float &meas_rate_min, float &meas_rate_max);
     void twitching_measure_acceleration(float &rate_of_change, float rate_measurement, float &rate_measurement_max);
-    void updating_rate_d_up(float &tune_d, float tune_d_min, float tune_d_max, float tune_d_step_ratio, float &tune_p, float tune_p_min, float tune_p_max, float tune_p_step_ratio, float rate_target, float meas_rate_min, float meas_rate_max);
-    void updating_rate_d_down(float &tune_d, float tune_d_min, float tune_d_step_ratio, float &tune_p, float tune_p_min, float tune_p_max, float tune_p_step_ratio, float rate_target, float meas_rate_min, float meas_rate_max);
-    void updating_rate_p_up_d_down(float &tune_d, float tune_d_min, float tune_d_step_ratio, float &tune_p, float tune_p_min, float tune_p_max, float tune_p_step_ratio, float rate_target, float meas_rate_min, float meas_rate_max);
-    void updating_angle_p_down(float &tune_p, float tune_p_min, float tune_p_step_ratio, float angle_target, float meas_angle_max, float meas_rate_min, float meas_rate_max);
-    void updating_angle_p_up(float &tune_p, float tune_p_max, float tune_p_step_ratio, float angle_target, float meas_angle_max, float meas_rate_min, float meas_rate_max);
+
+    // Added generic twitch test functions for multi
+    void twitch_test_init();
+    void twitch_test_run(uint8_t test_axis, const float dir_sign);
+
+    // replace multi specific updating gain functions with generic forms that covers all axes
+    virtual void updating_rate_p_up_all(uint8_t test_axis)=0;
```

Comment:
I think it would be good to add some comments above each of these methods (both in the .h and .cpp) to specify which TuneType they are used for.  This is partially an existing issue because AC_AutoTune.h is very sparse on comments in the lower half of the file but clearly that slipped by me in an earlier review.  For these new methods, let's do the right thing and make it more clear when they are used.

Reply:
done. thanks!